pkg/controller/namespaces/reconciler.go

```diff
@@ -123,6 +124,11 @@ func (n *NamespaceReconciler) UpsertNamespace(ctx context.Context, fragment *con
 	if fragment.Annotations != nil {
 		annotations = convertMap(fragment.Annotations)
 	}
+	if fragment.Service != nil && fragment.Service.SyncConfig != nil && fragment.Service.SyncConfig.NamespaceMetadata != nil {
```

Comment:
This isn't actually supposed to be on the namespace reconciler, it's meant for the service reconciler, where we upsert a namespace (although we can add it here too just a bit duplicative)

Reply:
Also we should respect `createNamespace` there too (if it's false)